pkg/apply/taskrunner/task.go

```diff
@@ -77,8 +78,25 @@ type WaitTask struct {
 	mu sync.RWMutex
 }
 
+// Equal returns true if the public fields are equal (private fields ignored).
+// Implements the "(T) Equal(T) bool" interface for cmp.Equal:
+// https://pkg.go.dev/github.com/google/go-cmp/cmp#Equal
+func (w *WaitTask) Equal(other *WaitTask) bool {
```

Comment:
I'm fine with adding such method to testing code, but I think it'd be better to use a [comparer](https://pkg.go.dev/github.com/google/go-cmp/cmp#Comparer) for `WaitTask`. Ideally production code shouldn't have any scaffolding for testing that otherwise would not be there.

Reply:
huh, i didn't know comparers existed.

The problem with the AssertEqual function interface is that it ends with a vararg, making it hard to add optional cmpopts to the EqualMatcher it wraps... I could use the EqualMatcher directly, but that makes it several more lines of code to use. I considered having a global variable to set default cmpopts for the Equalmatcher, but that just seems sloppy.

If you can think of a clean way to make the options optional, I'm all ears.

Reply:
Alright, I came up with a way to handle options with an optional parent Asserter struct... 

It's a tad confusing to look at, but allows the existing global functions to delegate to a global DefaultAsserter. Then if you need to use custom options you have to construct your own Asserter and use its methods instead of the global functions.